pkg/operator/controller/dns/controller.go

```diff
@@ -583,25 +583,24 @@ func (r *reconciler) createDNSProvider(dnsConfig *configv1.DNS, platformStatus *
 				return nil, err
 			}
 		} else if platformStatus.IBMCloud.DNSInstanceCRN != "" {
-			matches := ibm.IBMResourceCRNRegexp.FindStringSubmatch(platformStatus.IBMCloud.DNSInstanceCRN)
-			if len(matches) <= 0 {
-				return nil, fmt.Errorf("CRN does not match expected format: %s", platformStatus.IBMCloud.DNSInstanceCRN)
-			}
-			dnsProvider, err = getIbmDNSProvider(dnsConfig, creds, matches[ibm.IBMResourceCRNRegexp.SubexpIndex("guid")], userAgent, false)
+			dnsProvider, err = getIbmDNSProvider(dnsConfig, creds, platformStatus.IBMCloud.DNSInstanceCRN, userAgent, false)
 			if err != nil {
 				return nil, err
 			}
 		}
 	case configv1.PowerVSPlatformType:
 		//Power VS platform will use the ibm dns implementation
 		var err error
-		cisInstanceCRN := platformStatus.PowerVS.CISInstanceCRN
-		if cisInstanceCRN == "" {
-			return nil, fmt.Errorf("missing cis instance crn")
-		}
-		dnsProvider, err = getIbmDNSProvider(dnsConfig, creds, cisInstanceCRN, userAgent, true)
-		if err != nil {
-			return nil, fmt.Errorf("failed to create IBM DNS manager: %v", err)
+		if platformStatus.PowerVS.CISInstanceCRN != "" {
+			dnsProvider, err = getIbmDNSProvider(dnsConfig, creds, platformStatus.PowerVS.CISInstanceCRN, userAgent, true)
+			if err != nil {
+				return nil, err
+			}
+		} else if platformStatus.PowerVS.DNSInstanceCRN != "" {
+			dnsProvider, err = getIbmDNSProvider(dnsConfig, creds, platformStatus.PowerVS.DNSInstanceCRN, userAgent, false)
+			if err != nil {
+				return nil, err
+			}
```

Comment:
If `platformStatus.PowerVS.CISInstanceCRN` and `platformStatus.PowerVS.DNSInstanceCRN` are both empty, should `createDNSProvider` log something and return `&dns.FakeProvider{}`?  Come to think of it, the same question applies for `case configv1.IBMCloudPlatformType`.

Reply:
We expect either of this to be set by the installer but to be on the safer side  I have made changes to use the `&dns.FakeProvider{}` in case both of them are empty